tiledb/sm/array_schema/dimension.cc

```diff
@@ -1679,3 +1646,45 @@ Range DimensionVarSize::compute_mbr_var(
 }
 
 }  // namespace tiledb::sm
+
+std::string tile_extent_str(const tiledb::sm::Dimension& dim) {
+  std::stringstream ss;
+
+  if (!dim.tile_extent()) {
+    return tiledb::sm::constants::null_str;
+  }
+
+  auto g = [&](auto T) {
+    if constexpr (tiledb::type::TileDBNumeric<decltype(T)>) {
+      auto val = reinterpret_cast<const decltype(T)*>(dim.tile_extent().data());
+      ss << *val;
+      return ss.str();
+    }
+    throw std::logic_error(
+        "Datatype::" + datatype_str(dim.type()) + " not supported");
+    return std::string("");  // for return type deduction purposes
+  };
+  return apply_with_type(g, dim.type());
+}
+
+std::ostream& operator<<(std::ostream& os, const tiledb::sm::Dimension& dim) {
+  // Retrieve domain and tile extent strings
+  std::string domain_s = tiledb::type::range_str(dim.domain(), dim.type());
+  std::string tile_extent_s = tile_extent_str(dim);
```

Comment:
Looks like these local variables are only used once and could be inlined.

Reply:
`range_str()` is not used once